pkg/scd/operational_intents_handler.go

```diff
@@ -478,6 +478,36 @@ func checkUpsertPermissionsAndReturnManager(authorizedManager *api.Authorization
 	return dssmodels.Manager(*authorizedManager.ClientID), nil
 }
 
+// validateUpsertRequestAgainstPreviousOIRAndReturnVersion checks that the client requesting an OIR upsert has the necessary permissions and that the request is valid.
+// On success, the version of the OIR is returned:
+//   - upon initial creation (if no previous OIR exists), it is 0
+//   - otherwise, it is the version of the previous OIR
+func validateUpsertRequestAgainstPreviousOIRAndReturnVersion(
```

Comment:
nit: do not return version from this function and just perform validation.
Determining version can then just be:
```go
version := 0
if old != nil {
  version = int32(old.Version)
}
```